web/client/epics/longitudinalProfile.js

```diff
@@ -0,0 +1,477 @@
+                        changeMapInfoState(false),
+                        purgeMapInfoResults(),
+                        hideMapinfoMarker(),
+                        startDrawingAction
+                    )
+                    .takeUntil(action$.filter(({ type }) =>
+                        type === TOGGLE_MODE && dataSourceModeSelector(store.getState()) !== 'draw'
+                    ))
+                    .concat(deactivate());
+            case "select":
+                return Rx.Observable.from([
+                    purgeMapInfoResults(), hideMapinfoMarker(),
+                    ...(get(store.getState(), 'draw.drawOwner', '') === CONTROL_NAME ? DEACTIVATE_ACTIONS : []),
+                    registerEventListener('click', CONTROL_NAME),
+                    ...(mapInfoEnabledSelector(state) ? [toggleMapInfoState()] : [])
+                ]);
+            default:
+                return Rx.Observable.from([
+                    purgeMapInfoResults(), hideMapinfoMarker(),
+                    ...(get(store.getState(), 'draw.drawOwner', '') === CONTROL_NAME ? DEACTIVATE_ACTIONS : []),
+                    unRegisterEventListener('click', CONTROL_NAME)
+                ]);
+            }
+        });
+/**
+ * Reload chart data from WPS whenever geometry or request configuration changed
+ * @param action$
+ * @param store
+ * @returns {*}
+ */
+
+export const onChartPropsChangeEpic = (action$, store) =>
+    action$.ofType(CHANGE_GEOMETRY, CHANGE_DISTANCE, CHANGE_REFERENTIAL)
+        .filter(() => {
+            const state = store.getState();
+            return !isEmpty(geometrySelector(state));
+        })
+        .switchMap(() => {
+            const state = store.getState();
+            const geometry = geometrySelector(state);
+            const identifier = configSelector(state)?.identifier;
+            const wpsurl = configSelector(state)?.wpsurl;
+            const referential = configSelector(state)?.referential;
+            const distance = configSelector(state)?.distance;
+            const wpsBody = profileEnLong({identifier, geometry, distance, referential });
+            return executeProcess(wpsurl, wpsBody, {outputsExtractor: makeOutputsExtractor()})
+                .switchMap((result) => {
+                    const feature = {
+                        type: 'Feature',
+                        geometry
+                    };
+                    const map = mapSelector(state);
+                    const center = turfCenter(reprojectGeoJson(feature, geometry.projection, 'EPSG:4326')).geometry.coordinates;
+                    const bbox = turfBbox(reprojectGeoJson(feature, geometry.projection, 'EPSG:4326'));
+                    const [minx, minY, maxX, maxY] = bbox;
+                    const { infos, profile: points } = result ?? {};
+                    const styledFeatures = styleFeatures([feature], omit(highlightStyleSelector(state), ["radius"]));
```

Comment:
Note: in general it will be better to use the new styling format and to not apply a style object for each feature

Reply:
fixed, this was a copy pasta from old code